File: Coarse5/ch01/java-example/src/main/java/org/example/ex05/Exam05.java
package org.example.ex05;


import java.util.List;
import java.util.function.Consumer;
import java.util.function.Predicate;
import java.util.stream.Collectors;

public class Exam05 {

    private Predicate<String> stringPredicate = new Predicate<String>() {
        @Override
        public boolean test(String s) {
            return s.equals("?");
        }
    };

    public Exam05(){
        // 고차함수

        var strList = List.of(
            "1",
            "2",
            "홍길동",
            "함수",
            "메서드"
        );
        var result = strList.stream()
            .filter(
                (String s) -> s.equals("")
//                new Predicate<String>() {
//                    @Override
//                    public boolean test(String s) {
//                        return s.equals("");
//                    }
//                }

//                it -> it.equals("?")
            )
            .collect(Collectors.toList());
    }

    public static void main(String[] args) {
        new Exam05();
    }
}

File: Coarse5/ch01/kotlin-example/src/main/kotlin/ex05/Exam05.kt
package org.example.ex05

import java.util.function.Predicate


fun main(){
    var numberList = listOf<Int>(1,2,3,4,5)

    var pred = Predicate<Int> { t -> t%2 == 0 }

    numberList.stream().filter(pred)

    val add = { x: Int, y: Int -> x+y }

    var _add = fun(x:Int, y:Int): Int{
        return x+y
    }

    println(add(2,3))
    println(_add(3,4))

    lambda(4,5,_add)

//    numberList.filter { pred }
}

fun lambda(x:Int, y:Int, method: (Int,Int) -> Int){
    println(method(x,y))
}